flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/functions/hive/conversion/HiveInspectors.java

```diff
@@ -119,9 +141,45 @@ public static HiveObjectConversion getConversion(ObjectInspector inspector) {
 				} else {
 					return IdentityConversion.INSTANCE;
 				}
+			} else if (inspector instanceof JavaDateObjectInspector) {
+				if (((JavaDateObjectInspector) inspector).preferWritable()) {
```

Comment:
Nit: preferWritable() is defined in PrimitiveObjectInspector. So, in all those if-else if branches, we don't need those casts if we just cast inspector to PrimitiveObjectInspector once on the top.

Reply:
After doing some research,I think actually there is no need to check `preferWritable()`at all given input will be of primitive typed inspector. I will have a immediate following PR to take care of this

Reply:
okay. you're probably right.